azurelinuxagent/common/utils/archive.py

```diff
@@ -63,34 +69,49 @@ def __init__(self, lib_dir):
                     logger.error("{0} : {1}", self._source, e.strerror)
 
     def flush(self, timestamp):
-        files = self._get_files_to_archive()
-        if len(files) == 0:
+        copy_files, move_files = self._get_files_to_archive()
```

Comment:
The archive logic is called if the `update_goal_state` is not forced and if the incarnation changes. If the FastTrack extension config is modified then we might not capture it in our archive logic (which helps a lot with debugging). Before deleting the files in `save_config in wire.py (line:2012)`, you should call this function to make sure that we're capturing the FastTrackExtensionConfig too before deleting the files.